test/test-coordinator.js

```diff
@@ -95,42 +95,43 @@ require('util').inherits(TestCoordinator, events.EventEmitter);
 
 TestCoordinator.prototype.lookup = function(key) {
     var self = this;
-
-    var matchingNode;
-    var matchingHash;
-
     var hash = farmhash.fingerprint32(key);
 
-    function isBetterMatch(newHash) {
-        if (!matchingNode) return true; // 1 node is better than no node
-
-        if (hash <= newHash && newHash < matchingHash) return true; // new hash is closer to the current best match
-        if (matchingHash <= hash && hash <= newHash) return true; // the matchingHash wraps around, and new hash is a better match to hash
-
-        return false;
-    }
-
     var hostPorts = this.fakeNodes.map(function (fn) {
         return fn.getHostPort();
     });
     hostPorts.push(this.sutHostPort);
 
     // iterate over all hosts
+    var allHashes = [];
```

Comment:
I agree that it's pretty hard to follow but I'm not sure if you're version is the most readable option. I'll try to restructure this method again to try to improve it even further.